hass-addon-sunsynk-dev/run.py

```diff
@@ -208,6 +240,29 @@ async def read_sensors(
     return False
 
 
+async def write_sensor(filt: Filter, value) -> bool:
+    """Write sensor with the Modbus interface."""
+
+    sensor = filt.sensor
+    newv = ensure_tuple(value)
+    if newv == sensor.reg_value:
+        return False
+
+    _LOGGER.info(
+        "Writing sensor %s: %s=%s  [old %s]",
+        sensor.name,
+        sensor.id,
+        newv,
+        sensor.reg_value,
+    )
+    sensor.reg_value = newv
+    await SUNSYNK.write_sensor(sensor)
+    await read_sensors([sensor])
```

Comment:
Could this happen at the exact time as `poll_sensors()` and cause problems if we're trying to do two reads at the same time?

Reply:
Not with async code, you only have a single thread here. Read is only scheduled for the next second

The read might affect the filter (the filter makes the assumption a read only happens on the second). But these writable entities should all use the round-robin filter so its not a big issue